src/power_specials.c

```diff
@@ -552,22 +565,40 @@ void start_resurrect_creature(struct PlayerInfo *player, struct Thing *thing)
     }
 }
 
-TbBool create_transferred_creatures_on_level(void)
+long create_transferred_creatures_on_level(void)
 {
-    TbBool creature_created = false;
-    for (int i = 0; i < TRANSFER_CREATURE_STORAGE_COUNT; i++)
+    struct Thing* creatng;
+    struct Thing* heartng;
+    long creature_created = 0;
+    for (int p = 0; p < PLAYERS_COUNT; p++)
     {
-        if (intralvl.transferred_creatures[i].model > 0)
+        for (int i = 0; i < TRANSFER_CREATURE_STORAGE_COUNT; i++)
         {
-            struct Thing* thing = get_player_soul_container(my_player_number);
-            struct Coord3d* pos = &(thing->mappos);
-            thing = create_creature(pos, intralvl.transferred_creatures[i].model, 5);
-            if (thing_is_invalid(thing))
+            if (intralvl.transferred_creatures[p][i].model > 0)
             {
-                continue;
+                heartng = get_player_soul_container(p);
+                if (p == HERO_PLAYER)
+                {
+                    if (thing_is_invalid(heartng))
+                    {
+                        for (long n = 1; n < 16; n++)
+                        {
+                            heartng = find_hero_gate_of_number(n);
+                            if (!thing_is_invalid(heartng))
+                                break;
+                        }
+                    }
+                }
+
+                struct Coord3d* pos = &(heartng->mappos);
+                creatng = create_creature(pos, intralvl.transferred_creatures[p][i].model, 5);
```

Comment:
I believe 5 here means neutral, and if a player does it they would get claimed immediately, but I guess this wouldn't work for the hero player